client/verta/verta/_tracking/deployable_entity.py

```diff
@@ -35,6 +47,167 @@ def _histogram_endpoint(self):
+
+        Parameters
+        ----------
+        keys : list of str
+            Keys of artifacts to download.
+
+        Returns
+        -------
+        dict of str to str
+            Map of artifacts' keys to their cache filepaths—for use as the ``artifacts`` parameter
+            to a Verta class model.
+
+        Examples
+        --------
+        .. code-block:: python
+
+            run.log_artifact("weights", open("weights.npz", 'rb'))
+            # upload complete (weights)
+            run.log_artifact("text_embeddings", open("embedding.csv", 'rb'))
+            # upload complete (text_embeddings)
+            artifact_keys = ["weights", "text_embeddings"]
+            artifacts = run.fetch_artifacts(artifact_keys)
+            artifacts
+            # {'weights': '/Users/convoliution/.verta/cache/artifacts/50a9726b3666d99aea8af006cf224a7637d0c0b5febb3b0051192ce1e8615f47/weights.npz',
+            #  'text_embeddings': '/Users/convoliution/.verta/cache/artifacts/2d2d1d809e9bce229f0a766126ae75df14cadd1e8f182561ceae5ad5457a3c38/embedding.csv'}
+            ModelClass(artifacts=artifacts).predict(["Good book.", "Bad book!"])
+            # [0.955998517288053, 0.09809996313422353]
+            run.log_model(ModelClass, artifacts=artifact_keys)
+            # upload complete (custom_modules.zip)
+            # upload complete (model.pkl)
+            # upload complete (model_api.json)
+
+        """
+        if not (isinstance(keys, list)
+                and all(isinstance(key, six.string_types) for key in keys)):
+            raise TypeError("`keys` must be list of str, not {}".format(type(keys)))
+
+        # validate that `keys` are actually logged
+        self._refresh_cache()
+        existing_artifact_keys = {artifact.key for artifact in self._msg.artifacts}
+        unlogged_artifact_keys = set(keys) - existing_artifact_keys
+        if unlogged_artifact_keys:
+            raise ValueError("`keys` contains keys that have not been logged: {}".format(sorted(unlogged_artifact_keys)))
+
+        # get artifact checksums
+        paths = {artifact.key: artifact.path
+                 for artifact in self._msg.artifacts}
+
+        artifacts = dict()
+        for key in keys:
+            filename = os.path.join("artifacts", paths[key])
+
+            # check cache, otherwise write to cache
+            #     "try-get-then-create" can lead multiple threads trying to write to the cache
+            #     simultaneously, but artifacts being cached at a particular location should be
+            #     identical, so multiple writes would be idempotent.
+            path = self._get_cached_file(filename)
+            if path is None:
+                contents = self._get_artifact(key)
+                if isinstance(contents, tuple):  # ExperimentRun._get_artifact() returns (contents, path_only)
+                    contents, _ = contents  # TODO: raise error if path_only
```

Comment:
This was changed from 
```python
contents, _ = self._get_artifact(key)  # TODO: raise error if path_only
```
Because `ModelVersion._get_artifact()` only returns one value instead of `ExperimentRun`'s two, which is rather annoying and needs to be unified when these child classes are more thoroughly cleaned up.

Reply:
Can you note that in the code instead of a PR comment? This is very explanatory and will be 100% be lost (I see the comment, but that was more of an FYI than a FIXME explaining the issue.